runtime/runtime-params-estimator/src/cost.rs

```diff
@@ -7,12 +7,52 @@ use std::str::FromStr;
 #[derive(Copy, Clone, PartialEq, Eq, Debug, PartialOrd, Ord)]
 #[repr(u8)]
 pub enum Cost {
+    // Every set of actions in a transaction needs to be transformed into a
+    // action receipt, regardless of whether it is executed locally within the
+    // same block or delayed. The gas cost for this is paid when the receipt is
+    // created. The base amount is `action_receipt_creation_config.execution` +
+    // (either `action_receipt_creation.send_sir` or
+    // `action_receipt_creation.send_not_sir` depending on whether
+    // `sender == receiver`).
+    // On top of that, each type of action has its own costs defined, which is
+    // added for each action included in the receipt.
+    //
+    /// Estimates `ActionCosts::new_receipt`, which is the base cost for
+    /// creating a new action receipt, excluding actual action costs.
+    ///
+    /// Estimation: Measure the creation and execution of an empty action
+    /// receipt, where sender and receiver are two different accounts. This
+    /// involves applying two blocks and both contribute to the measured cost.
+    /// The cost is then divided by 2 to split between `send_not_sir` and
+    /// `execution` cost. `send_sir` is set to the same value as `send_not_sir`
+    /// for now.
     ActionReceiptCreation,
+    /// Estimates `ActionCosts::new_receipt`.`send_sir`. Although, note that it
+    /// is currently configured to be the same as `send_not_sir`. But we already
+    /// use this value as partial estimation of other action costs.
+    ///
+    /// Estimation: Measure the creation and execution of an empty action
+    /// receipt, where sender and receiver are the same account.
     ActionSirReceiptCreation,
     DataReceiptCreationBase,
     DataReceiptCreationPerByte,
     ActionCreateAccount,
+    // Deploying a new contract on the blockchain triggers a compilation process
+    // and the compiled code is stored in the DB. The costs charged for such a
+    // contract deployment action is `ActionDeployContractBase` + `N` *
+    // `ActionDeployContractPerByte`, where `N` is the number of bytes in the
+    // WASM code.
+    /// Estimates `action_creation_config.deploy_contract_cost`, which is
+    /// charged once per contract deployment
+    ///
+    /// Estimation: Measure deployment cost of a "smallest" contract.
```

Comment:
Let's rephrase this to highlight that:

* raw wasm is stored in a trie
* compiled code stored in a db